nucleus.go

```diff
@@ -6,26 +6,66 @@ package holochain
 
 import (
 	"fmt"
+	"github.com/google/uuid"
 	peer "github.com/libp2p/go-libp2p-peer"
 )
 
+type DNA struct {
+	Version                   int
+	UUID                      uuid.UUID
+	Name                      string
+	Properties                map[string]string
+	BasedOn                   Hash // references hash of another holochain that these schemas and code are derived from
+	RequiresVersion           int
+	DHTConfig                 DHTConfig
+	Progenitor                Progenitor
+	Zomes                     []Zome
+	PropertiesSchemaValidator SchemaValidator
```

Comment:
Storing the validator instead of the schema has the problem that EncodeDNA in GenChain can't include that schema's data as part of the hash.  I think we have to restore storing the actual properties schema here, and storing the validator just as a non-exported (lowercase) cached value.